alibi_detect/od/_gmm.py

```diff
@@ -0,0 +1,214 @@
+        batch_size
+            Batch size used to fit the detector. Only used if the ``'pytorch'`` backend is used. Defaults to ``None``.
+            If ``None``, the entire dataset is used in each epoch.
+        epochs
+            Number of epochs used to fit the detector. Used for both the ``'pytorch'`` and ``'sklearn'`` backends.
+            If the backend is ``'sklearn'``, the detector is fitted using the EM algorithm and the number of epochs
+            defualts to ``10``. If the backend is ``'pytorch'``, the detector is fitted using gradient descent and
+            the number of epochs defaults to ``100``.
+        tol
+            Tolerance used to fit the detector. Only used if the ``'sklearn'`` backend is used. Defaults to ``1e-3``.
+        n_init
+            Number of initializations used to fit the detector. Only used if the ``'sklearn'`` backend is used.
+            Defaults to ``1``.
+        init_params
+            Initialization method used to fit the detector. Only used if the ``'sklearn'`` backend is used. Must be
+            one of:
+            'kmeans' : responsibilities are initialized using kmeans.
+            'kmeans++' : responsibilities are initialized using kmeans++.
+            'random' : responsibilities are initialized randomly.
+            'random_from_data' : responsibilities are initialized randomly from the data.
+            Defaults to ``'kmeans'``.
+        verbose
+            Verbosity level used to fit the detector. Only used if the ``'sklearn'`` backend is used. Defaults to ``0``.
+        """
+        self.backend.fit(
+            self.backend._to_tensor(x_ref),
+            **self.backend.format_fit_kwargs(locals())
+        )
+
+    @catch_error('NotFittedError')
+    def score(self, x: np.ndarray) -> np.ndarray:
+        """Score `x` instances using the detector.
+
+        To score an instance, we compute the negative log-likelihood under the corresponding density function of
+        the fitted gaussian mixture model.
+
+        Parameters
+        ----------
+        x
+            Data to score. The shape of `x` should be `(n_instances, n_features)`.
+
+        Returns
+        -------
+        Outlier scores. The shape of the scores is `(n_instances,)`. The higher the score, the more anomalous the \
+        instance.
+
+        Raises
+        ------
+        NotFittedError
+            If called before detector has been fit.
+        """
+        score = self.backend.score(self.backend._to_tensor(x))
+        return self.backend._to_numpy(score)
+
+    @catch_error('NotFittedError')
+    def infer_threshold(self, x: np.ndarray, fpr: float) -> None:
+        """Infer the threshold for the GMM detector.
+
+        The threshold is computed so that the outlier detector would incorrectly classify `fpr` proportion of the
+        reference data as outliers.
+
+        Parameters
+        ----------
+        x
+            Reference data used to infer the threshold.
```

Comment:
Just seen this has been used across previous detectors as well, but can we remove "Reference" here given this data should be disjoint from the reference data used to fit the detectors